pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/design/AbstractCounterCheckRule.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.apex.rule.design;
+
+import static net.sourceforge.pmd.properties.constraints.NumericConstraints.positive;
+
+import java.lang.reflect.Modifier;
+
+import net.sourceforge.pmd.lang.apex.ast.AbstractApexNodeBase;
+import net.sourceforge.pmd.lang.apex.ast.ApexNode;
+import net.sourceforge.pmd.lang.apex.rule.AbstractApexRule;
+import net.sourceforge.pmd.lang.rule.internal.CommonPropertyDescriptors;
+import net.sourceforge.pmd.properties.PropertyDescriptor;
+
+
+/**
+ * Abstract class for rules counting some integer metric on some node.
+ *
+ * @author Clément Fournier
+ * @since 6.7.0
```

Comment:
6.14 now :P

Reply:
7.0.0 actually! my bad, just checked the reasons for targeting this for a major release